README.md

```diff
@@ -1 +1,98 @@
+
+1. This daemon runs as root. User must provide the path to the gateway configuration file (nvme_gateway.config) as an argument.
+
+2. To enable mTLS using self signed certificates, edit the nvme_gateway.config file to set:
+
+		enable_auth = True  # Setting this to False will open an insecure port
+		gateway_host = <IP address at which the client can reach the gateway>
+		gateway_port = <port at which the client can reach the gateway>
+
+3. Compiling protobuf files for grpc:
+
+		python3 -m grpc_tools.protoc --proto_path=./proto ./proto/nvme_gw.proto --python_out=. --grpc_python_out=.   
+
+4. RBD images needed to be exported as NVMe namespaces must exist
+
+5. SPDK v21.04 must be [checked out and set up](https://github.ibm.com/genctl-acadia/nvmeof-gateway#SPDK-Setup). Edit the nvme_gateway.config file to set:
+
+		spdk_path = <path to SPDK parent directory>
+		spdk_tgt = <path to SPDK target executable>
+
+
+# SPDK Setup
+
+	apt install librbd-dev
+
+	git clone https://github.com/spdk/spdk.git
+	git checkout v21.04
+	git submodule update --init
+
+Install dependencies:
+
+	./scripts/pkgdep.sh
+
+Intialize configuration:
+
+	./configure --with-rbd
+
+Build the SPDK app:
+
+        make
+
+Before running the SPDK target, setup huge pages:
+
+	sudo sh -c 'echo 4096 > /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages'
+
+
+# mTLS Configuration for testing purposes
+
+mTLS can be enabled by setting enable_auth=True in both the client & server scripts
+
+For testing purposes, self signed ertificates & keys can be generated locally using openssl:
+
+For the server, generate credentials for server name 'my.server' in files called server.key & server.crt
+
+  	% openssl req -x509 -newkey rsa:4096 -nodes -keyout server.key -out server.crt -days 3650 -subj '/CN=my.server'
+
+For client:
+
+  	% openssl req -x509 -newkey rsa:4096 -nodes -keyout client.key -out client.crt -days 3650 -subj '/CN=client1'
+
+Indicate the loaction of the keys and certificates in the nvme_gateway.conf file under the [MTLS] section
+
+
+# CLI Usage
+
+The CLI code can be used to initiate a connection to the gateway and run commands to configure the NVMe targets. Run the code with the -h flag to see a list of available commands:
+	
+	$ python3 ./nvme_gw_cli.py -h
+	usage: gwcli [-h] config_file {create_bdev,create_subsystem,create_namespace,allow_any_hosts,create_transport,create_listener} ...
+
+	Manage NVMe gateways
+
+	positional arguments:
+	  config_file           Gateway configuration file
+	  {create_bdev,create_subsystem,create_namespace,allow_any_hosts,create_transport,create_listener}
+
+	optional arguments:
+	  -h, --help            show this help message and exit
+
+Specify a command with the -h flag to see more information about the command, including a list of optional and required parameters:
+
+	$ python3 ./nvme_gw_cli.py nvme_gw.config create_bdev -h
+	usage: gwcli config_file create_bdev [-h] -r RBD -p POOL [-n NAME] [-u USER] [-b BLOCK_SIZE]
```

Comment:
Agreed - we plan on having a default location that the config file can be read from if the arg is not specified.

Reply:
Yes, then it should be an optional (e.g. `-f  config_file`), not positional, because you are not able to optionally specify a positional argument unless it is the last one. We can decide about a default location later, but why not make it not positional (i.e. specifying with `-f` option) from the start?

Reply:
> Yes, then it should be an optional (e.g. `-f config_file`), not positional, because you are not able to optionally specify a positional argument unless it is the last one. We can decide about a default location later, but why not make it not positional (i.e. specifying with `-f` option) from the start?

Agree. We'll change this.